Sources/NIO/System.swift

```diff
@@ -625,8 +625,25 @@ internal enum KQueue {
     @inline(never)
     @discardableResult
     public static func kevent(kq: CInt, changelist: UnsafePointer<kevent>?, nchanges: CInt, eventlist: UnsafeMutablePointer<kevent>?, nevents: CInt, timeout: UnsafePointer<Darwin.timespec>?) throws -> CInt {
+        var timeoutClamped = timeout
+        // Note that a nil timeout waits indefinitely
+        if let ts = timeout {
+            assert(ts.pointee.tv_nsec >= 0 || ts.pointee.tv_nsec < 1_000_000_000, "kevent timeout value is malformed (0 <= tv_nsec < 1_000_000_000)")
```

Comment:
This number seems a _little_ magic to me. Might be worth linking to some documents or something to say that `0 <= tv_nsec < 1_000_000_000`

Reply:
These are constraints on the `tv_nsec` field of the `timespec` datatype, which are latter checked in kernel. The `tv_nsec` field must hold a value [0, 999_999_999], I'll just add a ref to the C11 ISO standard timespec datatype.

Reply:
This condition is backwards and so all integers will pass. Negative timevals will be allowed (they will pass the second clause) and timevals greater than `1_000_000_000` will be allowed (they will pass the first clause).

We need a unit-test to validate that nothing crashes (it naturally doesn't need to wait for the task to complete, just confirm that the program crashes before the fix and doesn't crash after). That unit test would have caught this invalid assertion.

Separately, it'd be good to move this logic to closer to where `kevent` is invoked, rather than in the syscall wrapper. The syscall wrapper shouldn't paper over the idiosyncrasies of the call: the higher-level code can do that.

Finally, it may be easier to use `(0..<1_000_000_000).contains(ts.pointee.tv_nsec)` rather than try to figure out the right conditional.